crates/rpc/src/lib.rs

```diff
@@ -18,15 +19,14 @@ impl BeerusRpc {
         Self { beerus }
     }
 
-    pub async fn run(
-        self,
-    ) -> Result<(SocketAddr, ServerHandle), BeerusRpcError> {
+    pub async fn run(self) -> eyre::Result<(SocketAddr, ServerHandle)> {
```

Comment:
We will have `eyre` tightly coupled to our public API. 
I don't know if this is a good idea.

Reply:
Good point, we'll probably want to keep more informative errors.